api/envoy/admin/v2alpha/config_dump.proto

```diff
@@ -0,0 +1,118 @@
+
+  // The dynamically loaded warming listeners These are listeners that are currently undergoing
+  // warming in preparation to service data plane traffic.
+  repeated DynamicListener dynamic_warming_listeners = 4 [(gogoproto.nullable) = false];
+
+  // The dynamically loaded draining listeners These are listeners that are currently undergoing
+  // draining in preparation to stop servicing data plane traffic.
+  repeated DynamicListener dynamic_draining_listeners = 5 [(gogoproto.nullable) = false];
+}
+
+// Envoy's cluster manager fills this message with all currently known clusters.
+message ClustersConfigDump {
+  // This is the top level, "transactional," version information in the last processed CDS
+  // discovery response. If there are only static bootstrap clusters, this field will be "".
+  string version_info = 1;
+
+  // Describes a dynamically loaded cluster via the CDS API.
+  message DynamicCluster {
+    // This is the per-resource version information. This version is currently taken from the
+    // transactional version_info field at the time that the cluster was loaded. In the future,
+    // discrete per cluster versions may be supported by the API.
+    string version_info = 1;
+
+    // The cluster config.
+    envoy.api.v2.Cluster cluster = 2;
+  }
+
+  // The statically loaded cluster configs.
+  repeated envoy.api.v2.Cluster static_clusters = 2 [(gogoproto.nullable) = false];
+
+  // The dynamically loaded active clusters. These are clusters that are available to service
+  // data plane traffic.
+  repeated DynamicCluster dynamic_active_clusters = 3 [(gogoproto.nullable) = false];
+
+  // The dynamically loaded warming clusters. These are clusters that are currently undergoing
+  // warming in preparation to service data plane traffic.
+  repeated DynamicCluster dynamic_warming_clusters = 4 [(gogoproto.nullable) = false];
+}
+
+// Envoy's RDS implementation fills this message with all currently loaded routes, as described by
+// their RouteConfiguration objects. Static routes configured in the bootstrap configuration are
+// separated from those configured dynamically via RDS.
+message RoutesConfigDump {
+  message DynamicRouteConfig {
+    // This is the per-resource version information. This version is currently taken from the
+    // transactional version_info field in the RDS response at the time that the route was loaded.
+    // Note that *RoutesConfigDump* does not contain a top-level transactional *version_info*
+    // field as every RDS subscription is discrete.
+    string version_info = 1;
+
+    // The route config.
+    envoy.api.v2.RouteConfiguration route_config = 2;
+  }
+
+  // The statically loaded route configs.
+  repeated envoy.api.v2.RouteConfiguration static_route_configs = 2 [(gogoproto.nullable) = false];
+
+  // The dynamically loaded route configs.
+  repeated DynamicRouteConfig dynamic_route_configs = 3 [(gogoproto.nullable) = false];
+}
```

Comment:
No `ClusterLoadAssignment` (i.e. EDS) yet?

Reply:
I would like to think about adding that in a follow up, as we need to reconcile against the existing `/clusters` endpoint.